### ISSUE TITLE
HomeAssistantCore.update blocked from execution, no host internet connection

### ISSUE THREAD

**Issue by Jordan-Szwejda** [home-assistant/operating-system#3681]
### Describe the issue you are experiencing

I have been doing HA updates regularly. Unfortunately I cannot perform Home Assistant Core update to version 2024.11.2 from 2024.11.1.
I get information that :

"Failed to perform the action update/install. Error updating Home Assistant Core: 'HomeAssistantCore.update' blocked from execution, no host internet connection"

I would be grateful for any help to get updates working again. Everything else works as usual.

### What operating system image do you use?

generic-x86-64 (Generic UEFI capable x86-64 systems)

### What version of Home Assistant Operating System is installed?

Home Assistant OS 13.2

### Did the problem occur after upgrading the Operating System?

Yes

### Hardware details

Installed a few years ago on Proxmox. Nothing has been touched recently apart from installing HA updates.

### Steps to reproduce the issue

After clicking on web gui: Setting -> Home Assistant Core Update -> Update
I get "Failed to perform the action update/install. Error updating Home Assistant Core: 'HomeAssistantCore.update' blocked from execution, no host internet connection" message.

### Anything in the Supervisor logs that might be useful for us?

```txt
2024-11-18 22:51:43.662 INFO (MainThread) [__main__] Running Supervisor
2024-11-18 22:51:43.676 INFO (MainThread) [supervisor.os.manager] Rauc: slot A - activated slot kernel.0, marked slot kernel.0 as good
2024-11-18 22:51:43.681 INFO (MainThread) [supervisor.addons.manager] Phase 'initialize' starting 0 add-ons
2024-11-18 22:51:43.686 INFO (MainThread) [supervisor.addons.manager] Phase 'system' starting 1 add-ons
2024-11-18 22:51:44.625 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-mosquitto with version 6.4.1
2024-11-18 22:51:46.708 INFO (MainThread) [supervisor.services.modules.mqtt] Set core_mosquitto as service provider for mqtt
2024-11-18 22:51:49.627 INFO (MainThread) [supervisor.addons.manager] Phase 'services' starting 4 add-ons
2024-11-18 22:51:49.796 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-samba with version 12.3.2
2024-11-18 22:51:50.049 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/influxdb/amd64 with version 5.0.1
2024-11-18 22:51:50.097 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from core_samba
2024-11-18 22:51:50.356 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/grafana/amd64 with version 10.2.1
2024-11-18 22:51:50.659 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on ghcr.io/esphome/esphome-hassio with version 2024.10.3
2024-11-18 22:51:51.724 INFO (MainThread) [supervisor.api.middleware.security] /dns/info access from a0d7b954_influxdb
2024-11-18 22:52:04.285 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'mqtt-client'
2024-11-18 22:52:04.285 INFO (MainThread) [supervisor.auth] Home Assistant not running, checking cache
2024-11-18 22:52:20.379 INFO (MainThread) [supervisor.core] Start Home Assistant Core
2024-11-18 22:52:20.388 INFO (SyncWorker_1) [supervisor.docker.manager] Starting homeassistant
2024-11-18 22:52:20.500 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
2024-11-18 22:52:23.722 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state startup
2024-11-18 22:52:23.723 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
2024-11-18 22:52:25.545 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
2024-11-18 22:52:25.548 INFO (MainThread) [supervisor.homeassistant.core] Home Assistant Core state changed to APIState(core_state='NOT_RUNNING', offline_db_migration=False)
2024-11-18 22:52:29.029 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'mqtt-broker'
2024-11-18 22:52:29.032 INFO (MainThread) [supervisor.auth] Home Assistant not running, checking cache
2024-11-18 22:52:45.572 INFO (MainThread) [supervisor.homeassistant.core] Home Assistant Core state changed to APIState(core_state='STARTING', offline_db_migration=False)
2024-11-18 22:52:50.579 INFO (MainThread) [supervisor.homeassistant.core] Home Assistant Core state changed to APIState(core_state='RUNNING', offline_db_migration=False)
2024-11-18 22:52:50.579 INFO (MainThread) [supervisor.homeassistant.core] Detect a running Home Assistant instance
2024-11-18 22:52:50.579 INFO (MainThread) [supervisor.addons.manager] Phase 'application' starting 2 add-ons
2024-11-18 22:52:50.828 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on ghcr.io/sabeechen/hassio-google-drive-backup-amd64 with version 0.112.1
2024-11-18 22:52:51.058 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-piper with version 1.5.2
2024-11-18 22:52:51.615 INFO (MainThread) [supervisor.api.middleware.security] /core/info access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.618 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.620 INFO (MainThread) [supervisor.api.middleware.security] /mounts access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.621 INFO (MainThread) [supervisor.api.middleware.security] /addons access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.625 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.626 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.627 INFO (MainThread) [supervisor.api.middleware.security] /backups/9ab889bb/info access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.628 INFO (MainThread) [supervisor.api.middleware.security] /backups/e912ba3c/info access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.628 INFO (MainThread) [supervisor.api.middleware.security] /backups/66a5273d/info access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.629 INFO (MainThread) [supervisor.api.middleware.security] /backups/3efe5ef9/info access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.630 INFO (MainThread) [supervisor.api.middleware.security] /backups/84a3d105/info access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.631 INFO (MainThread) [supervisor.api.middleware.security] /backups/f479de8e/info access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:51.631 INFO (MainThread) [supervisor.api.middleware.security] /backups/aad910c1/info access from cebe7a76_hassio_google_drive_backup
2024-11-18 22:52:58.089 INFO (MainThread) [supervisor.misc.tasks] All core tasks are scheduled
2024-11-18 22:52:58.090 INFO (MainThread) [supervisor.core] Supervisor is up and running
2024-11-18 22:52:58.090 INFO (MainThread) [supervisor.host.info] Updating local host information
2024-11-18 22:52:58.090 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json
2024-11-18 22:52:58.093 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state running
2024-11-18 22:52:58.093 INFO (MainThread) [supervisor.resolution.checks.base] Run check for security/core
2024-11-18 22:52:58.093 INFO (MainThread) [supervisor.resolution.checks.base] Run check for ipv4_connection_problem/system
2024-11-18 22:52:58.093 INFO (MainThread) [supervisor.resolution.checks.base] Run check for trust/supervisor
2024-11-18 22:52:58.101 INFO (MainThread) [supervisor.resolution.checks.base] Run check for dns_server_ipv6_error/dns_server
2024-11-18 22:52:58.135 INFO (MainThread) [supervisor.resolution.checks.base] Run check for disabled_data_disk/system
2024-11-18 22:52:58.136 INFO (MainThread) [supervisor.resolution.checks.base] Run check for docker_config/system
2024-11-18 22:52:58.136 INFO (MainThread) [supervisor.resolution.checks.base] Run check for multiple_data_disks/system
2024-11-18 22:52:58.136 INFO (MainThread) [supervisor.resolution.checks.base] Run check for free_space/system
2024-11-18 22:52:58.136 INFO (MainThread) [supervisor.resolution.checks.base] Run check for dns_server_failed/dns_server
2024-11-18 22:52:58.171 INFO (MainThread) [supervisor.resolution.checks.base] Run check for no_current_backup/system
2024-11-18 22:52:58.171 INFO (MainThread) [supervisor.resolution.checks.base] Run check for pwned/addon
2024-11-18 22:52:58.301 INFO (MainThread) [supervisor.host.services] Updating service information
2024-11-18 22:52:58.303 INFO (MainThread) [supervisor.host.network] Updating local network information
2024-11-18 22:52:58.309 INFO (MainThread) [supervisor.resolution.check] System checks complete
2024-11-18 22:52:58.309 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state running
2024-11-18 22:52:58.347 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information
2024-11-18 22:52:58.353 INFO (MainThread) [supervisor.host.manager] Host information reload completed
2024-11-18 22:52:58.432 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
2024-11-18 22:52:58.432 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state running
2024-11-18 22:52:58.432 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
2024-11-18 22:55:59.559 WARNING (MainThread) [supervisor.jobs] 'HomeAssistantCore.update' blocked from execution, no host internet connection
```

### Anything in the Host logs that might be useful for us?

```txt
After running 'network info' i get:

host_internet: false
....
supervisor_internet: true

(I have problems with copying terminal output so I am getting it by screen recognition feature):

ha > network info
docker :
address: 172.30.32.0/23
dns: 172.30.32.3
gateway: 172.30.32.1
interface: hassio
host _ internet: false
i nterfaces :
connected: true
enabled: true
interface: enpOs18
ipv4 :
address :
- 192.168.0.101/24
gateway: 192.168.0.1
method: static
nameservers :
- 8.8.8.8
ready: true
ipv6 :
address :
- fe80:
gateway: null
nethod: disabled
nameservers: [ ]
ready: true
mac: <....>
primary: true
type: ethernet
ułan: null
wifi: null
superovisor_internet :true
```

## System Information

version | core-2024.11.1
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.4
os_name | Linux
os_version | 6.6.54-haos
arch | x86_64
timezone | Europe/Warsaw
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
HACS Data | ok
GitHub API Calls Remaining | 5000
Installed Version | 2.0.1
Stage | running
Available Repositories | 1539
Downloaded Repositories | 9

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 13.2
-- | --
update_channel | stable
supervisor_version | supervisor-2024.11.2
agent_version | 1.6.0
docker_version | 27.2.0
disk_total | 30.8 GB
disk_used | 20.8 GB
healthy | true
supported | true
host_connectivity | false
supervisor_connectivity | true
ntp_synchronized | true
virtualization | kvm
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Samba share (12.3.2), Let's Encrypt (5.2.3), InfluxDB (5.0.1), Grafana (10.2.1), Mosquitto broker (6.4.1), Home Assistant Google Drive Backup (0.112.1), Piper (1.5.2), ESPHome (2024.10.3)

</details>

<details><summary>Dashboards</summary>

dashboards | 5
-- | --
resources | 4
views | 13
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | November 8, 2024 at 21:52
-- | --
current_recorder_run | November 18, 2024 at 22:52
estimated_db_size | 410.51 MiB
database_engine | sqlite
database_version | 3.45.3

</details>

**Comment by Jordan-Szwejda** [comment]
I've just managed to resolve an issue with Home Assistant updates. The solution is rather unusual, and I suspect there might be a bug within the system. Here's a breakdown of the problem and my workaround:

To enable integration with Google Nest, I expose Home Assistant to my public IP address and use Let's Encrypt to generate valid SSL certificates periodically. This requires HTTP-based validation, so I forward port 80 from my router to port 8090 on the Home Assistant host, where Let's Encrypt is configured for HTTP challenges.

However, with this forwarding rule active, Home Assistant OS is unable to perform updates. As soon as I disable the rule and reboot the host, updates resume normally.

I hope this information proves helpful to others encountering similar update issues or provides insights for developers to identify and rectify this peculiar Home Assistant behavior.

**Comment by emerysteele** [comment]
Confirming same issue. Had port forwarding (80/HTTP) setup on my router to allow lets encrypt to work.

Did a bunch of stuff trying to troubleshoot. Even setup a new HAOS instance on a completely different machine with different IP address from my main instance. And had issues even on that machine trying to update the core, getting error: `'HomeAssistantCore.update' blocked from execution, no host internet connection`

Disabling the port 80 forwarding rule on my router & rebooting both HomeAssistant instances got updates working again.

Also seems to affect certain integrations. Specifically I was having issues with the TotalConnect integration.

**Comment by sairon** [comment]
That doesn't make any sense to me, in fact it makes me believe the problem is in the firewall configuration of the router. Can you @Jordan-Szwejda and @emerysteele share what router are you using and how you configure the forwarding rule?

The host internet test is performed by NetworkManager periodically (every 10 minutes) by doing an HTTP request to `http://checkonline.home-assistant.io/online.txt` - you can check if you can access it from HAOS, and from other machines on the network, when the firewall/forwarding rule is enabled. Forwarding port 80 of your router's WAN interface to a device in LAN should not affect _outgoing_ traffic on that port.

**Comment by Jordan-Szwejda** [comment]
I've been using MikroTik RouterOS 7.8 (https://www.mikrotik.com/). Here are the firewall rules I configured on the router. It's worth mentioning that I have a hairpin NAT rule set up. This allows me to access Home Assistant using the same domain address from both the external network (internet) and the local network. In both cases, I can use the same HTTPS URL for Home Assistant with a valid certificate.
[firewall_rules_mikrotik.txt](https://github.com/user-attachments/files/1
![rules](https://github.com/user-attachments/assets/fd78ff8d-470d-432d-a910-0182574626ae)
8013693/firewall_rules_mikrotik.txt)

Let me know if you need anything more.

**Comment by sairon** [comment]
@Jordan-Szwejda At first sight the rules are looking fine and AFAIK nothing changed recently that should change the behavior there, but you know, you can never rule out the butterfly effect :sweat_smile: Anyway the host internet connection is transparently taken from the NetworkManager and you can check `ha network info` if it currently thinks it's online. Anyway, if the issue comes back again, performing the checks I suggested could take us further.

**Comment by emerysteele** [comment]
I am using OPNSense.

Did some further troubleshooting, guess the port forwarding is a bit different than I'm used to. Thought it would only apply to inbound WAN traffic, but without setting specifically to 'WAN address' for destination, it also applies to outbound WAN traffic as well.

Guess I didn't notice cause most stuff is HTTPS/port 443 nowadays for web traffic.

![image](https://github.com/user-attachments/assets/fc9a58e8-7c44-4c8b-9281-76da6e305165)

**Comment by Jordan-Szwejda** [comment]
When I enable the forward rule then `network info` command starts returning 'host_internet: false' after a while whereas
`curl https://checkonline.home-assistant.io/online.txt` returns 'NetworkManager is online' response all the time.
Screenshot attached:

![image](https://github.com/user-attachments/assets/f37c2178-905a-4f8a-914a-f0e33a9e2069)

**Comment by emerysteele** [comment]
@Jordan-Szwejda try setting your In. Interface List to 'WAN' for the port 80 forwarding rule.

![image](https://github.com/user-attachments/assets/10012c2c-b086-47d8-aa99-58cc4d9657e2)

**Comment by Jordan-Szwejda** [comment]
@emerysteele Filtering 'In Interface' to WAN helps in this particular situation. Another thing is what really happens there that HAOS thinks there is no internet connection.
According to what @sairon said about NetworkManager there should not be broken internet connection reported due to the fact that I am able to retrieve correct response from http://checkonline.home-assistant.io/online.txt as it was presented on the screenshot in the previous comment.

**Comment by emerysteele** [comment]
You used https in your curl command which is port 443, not 80.

**Comment by Jordan-Szwejda** [comment]
Right, if I use http instead of https then I get an error. The reason why I used https is that I found this url in the supervisor repository defined for connectivity check: https://github.com/home-assistant/supervisor/blob/main/supervisor/supervisor.py

![image](https://github.com/user-attachments/assets/22bf91d8-a8d2-4e7f-ace0-2a1ee6479d56)

Anyway, that's a router configuration issue and I believe the thread can be closed. Thank you for participating in this thread and your help.
